extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampChannelExtractor.java

```diff
@@ -0,0 +1,167 @@
+// Created by Fynn Godau 2019, licensed GNU GPL version 3 or later
+
+package org.schabi.newpipe.extractor.services.bandcamp.extractors;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.jsoup.Jsoup;
+import org.schabi.newpipe.extractor.InfoItemsCollector;
+import org.schabi.newpipe.extractor.NewPipe;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.channel.ChannelExtractor;
+import org.schabi.newpipe.extractor.downloader.Downloader;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.exceptions.ParsingException;
+import org.schabi.newpipe.extractor.exceptions.ReCaptchaException;
+import org.schabi.newpipe.extractor.linkhandler.ListLinkHandler;
+import org.schabi.newpipe.extractor.stream.StreamInfoItem;
+import org.schabi.newpipe.extractor.stream.StreamInfoItemsCollector;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+public class BandcampChannelExtractor extends ChannelExtractor {
+
+    private JSONObject channelInfo;
+
+    public BandcampChannelExtractor(StreamingService service, ListLinkHandler linkHandler) throws ParsingException {
+        super(service, linkHandler);
+
+        channelInfo = getArtistDetails(getId());
+    }
+
+    /**
+     * Fetch artist details from mobile endpoint.
+     * <a href=https://notabug.org/fynngodau/bandcampDirect/wiki/rewindBandcamp+%E2%80%93+Fetching+artist+details>
+     * I once took a moment to note down how it works.</a>
+     */
+    public static JSONObject getArtistDetails(String id) throws ParsingException {
```

Comment:
Shouldn't be `public`

Reply:
I need to access this method in `BandcampChannelLinkHandlerFactory` to execute the same request in order to retrieve the URL from an ID.

Reply:
Then put it in `BandcampParsingHelper`. An `Extractor` just isn't the place for that

Reply:
Okay, I moved this method as well as `getImageUrl(long, boolean)` to the `BandcampExtractorHelper` class